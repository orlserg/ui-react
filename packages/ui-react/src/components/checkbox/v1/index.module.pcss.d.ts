export const checkbox: string
export const checkbox__description: string
export const checkbox__input: string
export const checkbox__input_indeterminate: string
export const checkbox_description: string
export const checkbox_disabled: string


interface Stylesheet {
    'checkbox': string
    'checkbox__description': string
    'checkbox__input': string
    'checkbox__input_indeterminate': string
    'checkbox_description': string
    'checkbox_disabled': string
}


declare const stylesheet: Stylesheet
export default stylesheet
