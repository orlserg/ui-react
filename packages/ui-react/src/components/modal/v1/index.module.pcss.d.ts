export const modal: string
export const modal__body: string
export const modal__body_bottom: string
export const modal__body_default: string
export const modal__body_reduced: string
export const modal__children: string
export const modal__children_bottom: string
export const modal__content: string
export const modal__content_l: string
export const modal__content_m: string
export const modal__content_s: string
export const modal__content_xl: string
export const modal__footer: string
export const modal__footer_bottom: string
export const modal_bottom: string
export const modal_center: string


interface Stylesheet {
    'modal': string
    'modal__body': string
    'modal__body_bottom': string
    'modal__body_default': string
    'modal__body_no-header': string
    'modal__body_reduced': string
    'modal__body_with-buttons': string
    'modal__children': string
    'modal__children_bottom': string
    'modal__content': string
    'modal__content_l': string
    'modal__content_m': string
    'modal__content_s': string
    'modal__content_xl': string
    'modal__footer': string
    'modal__footer_bottom': string
    'modal__header-wrapper': string
    'modal__icon-close': string
    'modal__icon-close_alone': string
    'modal__icon-close_default': string
    'modal_bottom': string
    'modal_center': string
}


declare const stylesheet: Stylesheet
export default stylesheet
