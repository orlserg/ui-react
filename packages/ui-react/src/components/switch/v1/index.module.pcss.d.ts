export const switch__input: string
export const switch_disabled: string


interface Stylesheet {
    'switch': string
    'switch__input': string
    'switch__input-view': string
    'switch_disabled': string
}


declare const stylesheet: Stylesheet
export default stylesheet
