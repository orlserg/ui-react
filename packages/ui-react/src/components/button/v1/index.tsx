import React, { ButtonHTMLAttributes, forwardRef, useMemo } from 'react';
import { isFragment, isElement } from 'react-is';

import { type TStyle, useClassnames } from '../../../hooks/use-classnames';
import { Loader } from '../../loader';

import style from './index.module.pcss';

type TNative = ButtonHTMLAttributes<HTMLButtonElement>;

export interface IProps {
    /**
     * CSS класс или классы, которые будут применены к кнопке.
     **/
    readonly className?: TStyle | string,
    /**
     * Устанавливает размер кнопки.
     **/
    readonly presetSize?: 'large' | 'medium' | 'small',
    /**
     * Устанавливает стиль кнопки.
     **/
    readonly presetStyle?: 'primary' | 'default' | 'dashed' | 'ghost' | 'negative' | 'success' | 'deprecated',
    /**
     * Если установлен в true, то кнопка будет в состоянии загрузки.
     **/
    readonly isLoading?: boolean,
    /**
     * Если установлен в true, то кнопка будет компактной.
     **/
    readonly isCompact?: boolean,
    /**
     * Уникальный идентификатор для кнопки.
     **/
    readonly id?: TNative['id'],
    /**
     * Устанавливает тип кнопки.
     **/
    readonly type?: TNative['type'],
    /**
     * Устанавливает порядок перехода по кнопке с помощью клавиши `Tab`.
     **/
    readonly tabIndex?: TNative['tabIndex'],
    /**
     * Определяет функцию обратного вызова, которая будет вызвана при фокусировке на кнопке.
     **/
    readonly onFocus?: TNative['onFocus'],
    /**
     * Определяет функцию обратного вызова, которая будет выполнена при нажатии на кнопку.
     **/
    readonly onClick?: TNative['onClick'],
    /**
     * Содержимое кнопки.
     **/
    readonly children?: TNative['children'],
    /**
     * Устанавливает горячую клавишу для кнопки.
     **/
    readonly accessKey?: TNative['accessKey'],
    /**
     * Если установлен в true, то кнопка будет заблокирована.
     **/
    readonly disabled?: TNative['disabled']
}

const LOADER_STYLE_MAP = {
    primary   : 'white',
    default   : 'rich-grey',
    dashed    : 'rich-grey',
    ghost     : 'rich-grey',
    negative  : 'red',
    success   : 'rich-grey',
    deprecated: 'white'
} as const;

const isValidCompact = (element: IProps['children']) => {
    // @ts-expect-error Непонятно как быть с типом
    return isElement(element) && !!(element.type === 'svg' || !isFragment(element) && element.type.name?.startsWith('Icon'));
};

const isCompact = (element: IProps['children']) => {
    return Array.isArray(element) ? element.every(isValidCompact) : isValidCompact(element);
};

/**
 * Компонент `Button` представляет собой кнопку, которая может использоваться для выполнения каких-либо действий в приложении.
 **/
export const Button = forwardRef<HTMLButtonElement | null, IProps>(({ presetSize = 'medium', presetStyle = 'default', type = 'button', ...props }, ref) => {
    const cn = useClassnames(style, props.className);

    const elChildren = useMemo(() => {
        if(props.isLoading) {
            return (
                <Loader
                    presetStyle={LOADER_STYLE_MAP[presetStyle]}
                    presetSize="button"
                />
            );
        }

        return props.children;
    }, [props.children, props.isLoading, presetStyle]);

    return (
        <button
            id={props.id}
            type={type}
            ref={ref}
            // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
            disabled={props.isLoading || props.disabled}
            // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
            tabIndex={props.isLoading || props.disabled ? -1 : props.tabIndex}
            onFocus={props.onFocus}
            onClick={props.onClick}
            accessKey={props.accessKey}
            className={cn('button', {
                [`button_${presetSize}`] : presetSize,
                [`button_${presetStyle}`]: presetStyle,
                'button_compact'         : typeof props.isCompact === 'undefined' ? isCompact(props.children) : props.isCompact
            })}
            children={elChildren}
        />
    );
});
