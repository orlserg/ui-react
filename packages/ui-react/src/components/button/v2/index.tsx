import React, { ButtonHTMLAttributes, forwardRef } from 'react';
import { isFragment, isElement } from 'react-is';

import { type TStyle, useClassnames } from '../../../hooks/use-classnames';

import style from './index.module.pcss';

type TNative = ButtonHTMLAttributes<HTMLButtonElement>;

export interface IProps {
    /**
     * Устанавливает связь с формой
     **/
    readonly form?: TNative['form'],
    /**
     * CSS класс или классы, которые будут применены к кнопке.
     **/
    readonly className?: TStyle | string,
    /**
     * Устанавливает размер кнопки.
     **/
    readonly presetSize?: 'medium' | 'small',
    /**
     * Устанавливает стиль кнопки.
     **/
    readonly presetStyle?: 'primary' | 'primary-circle' | 'secondary' | 'secondary-circle' | 'interface' | 'interface-circle' | 'error' | 'ghost' | 'ghost-error',
    /**
     * Если установлен в true, то кнопка будет в состоянии загрузки.
     **/
    readonly isLoading?: boolean,
    /**
     * Если установлен в true, то кнопка будет компактной.
     **/
    readonly isCompact?: boolean,
    /**
     * Уникальный идентификатор для кнопки.
     **/
    readonly id?: TNative['id'],
    /**
     * Устанавливает тип кнопки.
     **/
    readonly type?: TNative['type'],
    /**
     * Устанавливает порядок перехода по кнопке с помощью клавиши `Tab`.
     **/
    readonly tabIndex?: TNative['tabIndex'],
    /**
     * Определяет функцию обратного вызова, которая будет вызвана при фокусировке на кнопке.
     **/
    readonly onFocus?: TNative['onFocus'],
    /**
     * Определяет функцию обратного вызова, которая будет выполнена при нажатии на кнопку.
     **/
    readonly onClick?: TNative['onClick'],
    /**
     * Содержимое кнопки.
     **/
    readonly children?: TNative['children'],
    /**
     * Устанавливает горячую клавишу для кнопки.
     **/
    readonly accessKey?: TNative['accessKey'],
    /**
     * Если установлен в true, то кнопка будет заблокирована.
     **/
    readonly disabled?: TNative['disabled']
}

const isValidCompact = (element: IProps['children']) => {
    // @ts-expect-error Непонятно как быть с типом
    return isElement(element) && !!(element.type === 'svg' || !isFragment(element) && element.type.name?.startsWith('Icon'));
};

const isCompact = (element: IProps['children']) => {
    console.log('asd', element);

    return !Array.isArray(element) && isValidCompact(element);
};

/**
 * Компонент `Button` представляет собой кнопку, которая может использоваться для выполнения каких-либо действий в приложении.
 **/
export const Button = forwardRef<HTMLButtonElement | null, IProps>(({ presetSize = 'medium', presetStyle = 'primary', type = 'button', ...props }, ref) => {
    const cn = useClassnames(style, props.className);

    return (
        <button
            id={props.id}
            type={type}
            ref={ref}
            form={props.form}
            // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
            disabled={props.isLoading || props.disabled}
            // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
            tabIndex={props.isLoading || props.disabled ? -1 : props.tabIndex}
            onFocus={props.onFocus}
            onClick={props.onClick}
            accessKey={props.accessKey}
            className={cn('button', {
                [`button_${presetSize}`] : presetSize,
                [`button_${presetStyle}`]: presetStyle,
                'button_is-loading'      : props.isLoading,
                'button_compact'         : typeof props.isCompact === 'undefined' ? isCompact(props.children) : props.isCompact
            })}
            children={props.children}
        />
    );
});
