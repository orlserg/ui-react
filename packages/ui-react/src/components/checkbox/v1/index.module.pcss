.checkbox {
    display: grid;
    grid-gap: 4px 14px;
    color: var(--checkbox-label-color);
    font-family: var(--font-family-primary);
    font-style: normal;
    font-weight: 400;
    font-size: 16px;
    line-height: 20px;
    grid-auto-flow: column;
    justify-content: flex-start;

    --color-description: var(--color-text-secondary);
    --checkbox-label-color: var(--color-text-basic);

    &_disabled {
        color: var(--checkbox-label-color);

        --color-description: var(--color-text-disabled);
    }

    &_description {
        grid-auto-flow: initial;
        grid-template-areas: "a b" "a c";
    }

    @media (hover: hover) and (pointer: fine) {

        &:hover:not(&_disabled) {
            cursor: pointer;
        }
    }

    &__input {
        margin: 0;
        padding: 0;
        appearance: none;
        background-color: var(--color-bg-base);
        border: 1px solid var(--color-border-basic);
        width: 20px;
        height: 20px;
        box-sizing: border-box;
        border-radius: 4px;
        background-position: center center;
        background-repeat: no-repeat;
        transition: all 200ms linear;

        @media screen and (prefers-reduced-motion: reduce) {
            transition: none;
        }

        @media (hover: hover) and (pointer: fine) {

            &:hover:not([disabled]) {
                cursor: pointer;
                border-color: var(--color-bg-brand-2);
            }
        }

        &:checked,
        &_indeterminate {
            background-color: var(--color-bg-accent);
            border-color: var(--color-bg-accent);

            @media (hover: hover) and (pointer: fine) {

                &:hover:not([disabled]) {
                    opacity: 0.8;
                }
            }
        }

        &[disabled] {
            background-color: var(--color-bg-plate);
            border-color: var(--color-border-basic);
        }

        &:checked {
            background-image: url("./img/checkmark.svg");

            &[disabled] {
                background-image: url("./img/checkmark_disabled.svg");
            }
        }

        &_indeterminate,
        &_indeterminate:checked {
            background-image: url("./img/indeterminate.svg");

            &[disabled] {
                background-image: url("./img/indeterminate_disabled.svg");
            }
        }

        &:focus:not([disabled]) {
            outline: none;
            box-shadow: 0 0 0 2px var(--box-shadow-focus);
        }
    }

    &__description {
        color: var(--color-description);
        grid-area: c;
    }
}
