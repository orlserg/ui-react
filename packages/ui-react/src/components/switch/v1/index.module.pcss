.switch {
    display: grid;
    grid-auto-flow: column;
    grid-column-gap: 12px;
    place-items: center flex-start;
    place-content: center flex-start;

    --switch-background-color: var(--color-border-basic);
    --switch-background-color-hover: rgba(209, 207, 215, 0.8);

    @media (hover: hover) and (pointer: fine) {

        &:hover:not(&_disabled) {
            cursor: pointer;

            --switch-background-color: var(--switch-background-color-hover);
        }
    }

    &__input {
        display: none;
        opacity: 0;
        width: 0;
        height: 0;
    }

    &__input-view {
        box-sizing: border-box;
        border-radius: 8px;
        padding: 0;
        margin: 0;
        height: 16px;
        width: 28px;
        background-color: var(--switch-background-color);
        position: relative;
        transition: all 150ms linear;

        @media screen and (prefers-reduced-motion: reduce) {
            transition: none;
        }

        &::before {
            content: "";
            display: block;
            width: 14px;
            height: 14px;
            background-color: var(--color-bg-base);
            border-radius: 50%;
            position: absolute;
            top: 1px;
            left: 1px;
            box-shadow: -1px 0 2px rgba(0, 0, 0, 0.08);
            transition: all 150ms linear;

            @media screen and (prefers-reduced-motion: reduce) {
                transition: none;
            }
        }
    }

    &__input:checked + &__input-view {
        --switch-background-color: var(--color-bg-brand);

        &::before {
            left: calc(100% - 15px);
        }
    }

    &__input[disabled] + &__input-view {

        &::before {
            opacity: 0.6;
        }
    }

    &__input:checked[disabled] + &__input-view {
        opacity: 0.4;
    }

    &__input:not([disabled]) + &__input-view:focus {
        outline: none;
        box-shadow: 0 0 0 2px rgba(80, 103, 222, 0.2);
    }
}
