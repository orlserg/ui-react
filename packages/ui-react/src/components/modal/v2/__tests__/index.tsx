import React, { Fragment } from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react';

import { Modal } from '../index';
import { Button } from '../../../button/v1';

test('Default use', async () => {
    let isOpen = false;
    const { asFragment, getByText } = render(
        <Fragment>
            <Button onClick={() => isOpen = true}>Trigger</Button>
            <Modal
                header="Modal header"
                isOpen={isOpen}
                onClose={() => isOpen = false}
                primaryButton={{
                    children: 'Primary button'
                }}
                secondaryButton={{
                    children: 'Secondary button',
                    onClick : () => isOpen = false
                }}
            >
                Modal content
            </Modal>
        </Fragment>
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Single Button', async () => {
    let isOpen = false;
    const { asFragment, getByText } = render(
        <Fragment>
            <Button onClick={() => isOpen = true}>Trigger</Button>
            <Modal
                header="Modal header"
                isOpen={isOpen}
                widthPreset="m"
                onClose={() => isOpen = false}
                primaryButton={{
                    children: 'Primary button'
                }}
            >
                Modal content
            </Modal>
        </Fragment>
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Placement bottom', async () => {
    let isOpen = false;
    const { asFragment, getByText } = render(
        <Fragment>
            <Button onClick={() => isOpen = true}>Trigger</Button>
            <Modal
                placement="bottom"
                widthPreset="xl"
                header="Modal header"
                isOpen={isOpen}
                onClose={() => isOpen = false}
                primaryButton={{
                    children: 'Primary button'
                }}
            >
                Modal content
            </Modal>
        </Fragment>
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Placement bottom', async () => {
    let isOpen = false;
    const { asFragment, getByText } = render(
        <Fragment>
            <Button onClick={() => isOpen = true}>Trigger</Button>
            <Modal
                widthPreset="s"
                header="Modal header"
                isOpen={isOpen}
                onClose={() => isOpen = false}
                primaryButton={{
                    children: 'Primary button'
                }}
            >
                Modal content
            </Modal>
        </Fragment>
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Esc press', async () => {
    let isOpen = false;
    const { asFragment, getByText } = render(
        <Fragment>
            <Button onClick={() => isOpen = true}>Trigger</Button>
            <Modal
                widthPreset="s"
                header="Modal header"
                isOpen={isOpen}
                onClose={() => isOpen = false}
                primaryButton={{
                    children: 'Primary button'
                }}
            >
                Modal content
            </Modal>
        </Fragment>
    );

    fireEvent.keyDown(getByText('Trigger'), {
        key: 'Escape',
        code: 'Escape',
        keyCode: 27,
        charCode: 27
    });

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Layout click', async () => {
    const handleOpen = jest.fn();
    const handleClose = jest.fn();

    const { asFragment, getByText, getByRole } = render(
        <Fragment>
            <Button onClick={handleOpen}>Trigger</Button>
            <Modal
                widthPreset="s"
                header="Modal header"
                isOpen={true}
                onClose={handleClose}
                primaryButton={{
                    children: 'Primary button'
                }}
            >
                Modal content
            </Modal>
        </Fragment>
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Trigger'));

    const dialog = getByRole('dialog');

    fireEvent.click(dialog);

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('Header node', async () => {
    const handleOpen = jest.fn();
    const handleClose = jest.fn();

    const { asFragment, getByText, getByRole } = render(
        <Fragment>
            <Button onClick={handleOpen}>Trigger</Button>
            <Modal
                widthPreset="s"
                header={(
                    <div>Modal header</div>
                )}
                isOpen={true}
                onClose={handleClose}
                primaryButton={{
                    children: 'Primary button'
                }}
            >
                Modal content
            </Modal>
        </Fragment>
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Trigger'));

    const dialog = getByRole('dialog');

    fireEvent.click(dialog);

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('isCloseIconShow', async () => {
    const handleOpen = jest.fn();
    const handleClose = jest.fn();

    const { asFragment, getByText, getByRole } = render(
        <Fragment>
            <Button onClick={handleOpen}>Trigger</Button>
            <Modal
                widthPreset="s"
                isCloseIconShow={true}
                isOpen={true}
                onClose={handleClose}
                primaryButton={{
                    children: 'Primary button'
                }}
            >
                Modal content
            </Modal>
        </Fragment>
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Trigger'));

    const dialog = getByRole('dialog');

    fireEvent.click(dialog);

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('isOpen false no buttons', async () => {
    const handleOpen = jest.fn();
    const handleClose = jest.fn();

    const { asFragment, getByText } = render(
        <Fragment>
            <Button onClick={handleOpen}>Trigger</Button>
            <Modal
                widthPreset="l"
                isCloseIconShow={true}
                onClose={handleClose}
                bodyPaddingPreset="reduced"
            >
                Modal content
            </Modal>
        </Fragment>
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});

test('isOpen false no buttons', async () => {
    const handleOpen = jest.fn();
    const handleClose = jest.fn();

    const { asFragment, getByText } = render(
        <Fragment>
            <Button onClick={handleOpen}>Trigger</Button>
            <Modal
                onClose={handleClose}
                secondaryButton={{
                    children: 'Button',
                    onClick : handleClose
                }}
            >
                <div>Modal content</div>
            </Modal>
        </Fragment>
    );

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(getByText('Trigger'));

    await waitFor(() => {
        expect(asFragment()).toMatchSnapshot();
    });
});
