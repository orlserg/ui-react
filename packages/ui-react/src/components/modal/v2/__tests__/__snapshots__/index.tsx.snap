// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default use 1`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`Default use 2`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`Esc press 1`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`Esc press 2`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`Header node 1`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`Header node 2`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`Layout click 1`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`Layout click 2`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`Placement bottom 1`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`Placement bottom 2`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`Placement bottom 3`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`Placement bottom 4`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`Single Button 1`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`Single Button 2`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`isCloseIconShow 1`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`isCloseIconShow 2`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`isOpen false no buttons 1`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`isOpen false no buttons 2`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`isOpen false no buttons 3`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;

exports[`isOpen false no buttons 4`] = `
<DocumentFragment>
  <button
    class="button button_medium button_default"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;
